### ISSUE TITLE
Workers - Azure Cloud Service storage authentication failure

### ISSUE THREAD

**Issue by thebigmoosey** [OctopusDeploy/Issues#5177]
# Prerequisites

- [x] I have verified the problem exists in the latest version
- [x] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [x] I have written a descriptive issue title
- [x] I have linked the original source of this report
- [x] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug

When attempting to deploy an Azure Cloud Service using the Default Worker Pool, using a classic storage account, and a CS target that passes a health check, we receive the following authentication error regarding storage accounts.

NOTE:
**We believe this has to do with the permissions around the worker account, still under investigation.**

### Log excerpt

Example failure task log (for Octopus Internal Staff): https://deploy.octopushq.com/app#/projects/mark-siedle-s-test-project-pls-don-t-remove/releases/1.0.37/deployments/Deployments-43150

```
23:57:35   Error    |       ForbiddenError: The server failed to authenticate the request. Verify that the certificate is valid and is associated with this subscription.
23:57:35   Error    |       Hyak.Common.CloudException
23:57:35   Error    |       at Microsoft.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
23:57:35   Error    |       at Microsoft.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccess(Task task)
23:57:35   Error    |       at Microsoft.WindowsAzure.Management.Storage.StorageAccountOperationsExtensions.GetKeys(IStorageAccountOperations operations, String accountName)
23:57:35   Error    |       at Calamari.Azure.Integration.AzurePackageUploader.GetStorageAccountPrimaryKey(SubscriptionCloudCredentials credentials, String storageAccountName, String serviceManagementEndpoint)
23:57:35   Error    |       at Calamari.Azure.Integration.AzurePackageUploader.Upload(SubscriptionCloudCredentials credentials, String storageAccountName, String packageFile, String uploadedFileName, String storageEndpointSuffix, String serviceManagementEndpoint)
23:57:35   Error    |       at Calamari.Azure.Deployment.Conventions.UploadAzureCloudServicePackageConvention.Install(RunningDeployment deployment)
23:57:35   Error    |       at Calamari.Deployment.ConventionProcessor.RunInstallConventions()
23:57:35   Error    |       at Calamari.Deployment.ConventionProcessor.RunConventions()
23:57:35   Error    |       at Calamari.Azure.Commands.DeployAzureCloudServiceCommand.Execute(String[] commandLineArguments)
23:57:35   Error    |       at Calamari.Program.Execute(String[] args)
23:57:36   Verbose  |       Process C:\Windows\system32\WindowsPowershell\v1.0\PowerShell.exe in D:\Octopus\Work\20181218235702-363433-2984 exited with code 100
```

## Workarounds

There are no workarounds known at this time.

## Links

Source: https://help.octopus.com/t/cloud-service-deployment-fails-with-authentication-error/21981

**Comment by shaunmarx** [comment]
We have managed to reproduce this on a single machine, however we have not been able to reproduce this locally or on another machine even with the exact same configuration as where we did manage to reproduce it. We are unable to explain the exact cause for failure and trying to work out how to best get to the bottom of this one.

So far we have looked at the permissions, certificate stores, certificates, crypto settings i.e. TLS levels and enabled cipher suites none of which seem to be the issue in this case.
